docs/docs/articles/run-tests-with-github-actions.md

```diff
@@ -0,0 +1,143 @@
+
+## Usage
+To use the action in your GitHub workflow, use kubeshop/testkube-run-action@v1 action. The configuration options are described in the Inputs section and may vary depending on the Testkube solution you are using (cloud or self-hosted) and your needs.
+
+The most important options you will need are test and testSuite - you should pass a test or test suite name there.
+
+### Testkube Cloud
+To use this GitHub Action for the Testkube Cloud, you need to create an API token.
+
+Then, pass the **organization** and **environment** IDs for the test, along with the **token** and other parameters specific for your use case:
+
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  # Instance
+  organization: tkcorg_0123456789abcdef
+  environment: tkcenv_fedcba9876543210
+  token: tkcapi_0123456789abcdef0123456789abcd
+  # Options
+  test: some-test-id-to-run
+  ```
+
+It will probably be unsafe to keep this directly in the workflow's YAML configuration, so you may want to use [GitHub's secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) instead.
+
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  # Instance
+  organization: ${{ secrets.TkOrganization }}
+  environment: ${{ secrets.TkEnvironment }}
+  token: ${{ secrets.TkToken }}
+  # Options
+  test: some-test-id-to-run
+  ```
+
+### Self-hosted Instance
+
+
+To run the test on self-hosted instance, simply pass url that points to the API, i.e.:
+
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  # Instance
+  url: https://demo.testkube.io/results/v1
+  # Options
+  test: some-test-id-to-run
+  ```
+
+### Examples
+
+Run a test on a self-hosted instance:
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  url: https://demo.testkube.io/results/v1
+  test: container-executor-curl-smoke
+  ```
+Run a test suite on a self-hosted instance:
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  url: https://demo.testkube.io/results/v1
+  testSuite: executor-soapui-smoke-tests
+  ```
+Run tests from a local repository for the PR:
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  organization: ${{ secrets.TkOrganization }}
+  environment: ${{ secrets.TkEnvironment }}
+  token: ${{ secrets.TkToken }}
+  test: e2e-dashboard-tests
+  ref: ${{ github.head_ref }}
+  ```
+
+Run tests with custom environment variables:
+
+```sh
+uses: kubeshop/testkube-run-action@v1
+with:
+  organization: ${{ secrets.TkOrganization }}
+  environment: ${{ secrets.TkEnvironment }}
+  token: ${{ secrets.TkToken }}
+  test: e2e-dashboard-tests
+  variables: |
+    URL="https://some.domain.com"
+    EXECUTED_FROM="${{ github.head_ref }}"
+  secretVariables: |
+    SOME_SECRET_ENV="abc"
+    OTHER_ENV="${{ secrets.ExternalToken }}"
+```
+
+## Inputs
+There are different inputs available for tests and test suites, as well as for Cloud and your own instance.
+
+### Test
+
+Required	Name	Description
```

Comment:
Shouldn't it be formatted as a code or a table?

Reply:
Yes - thanks for the catch - worked on these but apparently didn't save - I'll correct it.